perma_web/perma/forms.py

```diff
@@ -187,6 +187,19 @@ def make_random_password(len=20):
         user = forms.ModelForm.save(self, commit)
         return user
 
+    def user_is_logged_in(self) -> bool:
+        """Determine whether there is a currently logged-in user.
+
+        This may be useful for determining, e.g., whether or not to
+        display certain user registration fields.
+        """
+        return (
+            hasattr(self, 'request')
+            and hasattr(self.request, 'user')
+            and isinstance(self.request.user, LinkUser)
+        )
```

Comment:
Good catch! This was an oversight on my part. I just pushed up [48ef889](https://github.com/harvard-lil/perma/pull/3631/commits/48ef889e8682461898db5e6d80dcd8b92642476d), which tweaks `UserForm.user_is_logged_in` to use `is_authenticated`.

(It also now requires the `request` to be passed in explicitly. Otherwise, there might someday be a confusing condition where `user_is_logged_in()` returns false not because the user is not logged in, but because we forgot to set `request` as an attribute of the form.)